### PR TITLE
gdcm: add 3.0.20

### PR DESCRIPTION
**Pull request by Nekto89**
Use system openssl, json-c, libiconv, libuuid

Specify library name and version:  **gdcm/all**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by uilianries — APPROVED** [review]
LGTM

**Review by prince-chrismc — APPROVED** [review]

File: recipes/gdcm/all/conanfile.py [hunk 1]
```diff
@@ -69,8 +81,12 @@ def generate(self):
         tc.variables["GDCM_BUILD_SHARED_LIBS"] = bool(self.options.shared)
         # FIXME: unvendor deps https://github.com/conan-io/conan-center-index/pull/5705#discussion_r647224146
         tc.variables["GDCM_USE_SYSTEM_EXPAT"] = True
+        tc.variables["GDCM_USE_SYSTEM_JSON"] = self.options.with_json
```
Comment (SpaceIm) [inline recipes/gdcm/all/conanfile.py:84]: I don't understand. What happen when with_json option is False? gdcm still uses json-c but bundle one?

Comment (SpaceIm) [reply recipes/gdcm/all/conanfile.py:84]: Moreover the json patch could be more robust, upstream doesn't rely on official config file.

Comment (Nekto89) [reply recipes/gdcm/all/conanfile.py:84]: If I understood correctly, without GDCM_USE_SYSTEM_JSON json functionality just won't work. Same with openssl. Theoretically some people might want to use it but it's not part of core functionality - I didn't even know that gdcm has it for many years.

**Pull request merged**